### PR TITLE
feat(bench): dense-matching benchmark harness + amend ADR 0015 (C5)

### PR DESCRIPTION
**Pull request by VitalyVorobyev**
## What

Resets the C5 dense-matcher direction (user-supervised) and lands the **benchmark harness** the matcher will be scored against.

## ADR 0015 amended

The original ceiling forbade an in-house matcher and mandated wrapping `opencv-rust` SGBM behind a feature flag. **Reversed** (dated amendment note in the ADR): the production stack is Rust-native, so the matcher ships **pure-Rust in `vision-mvg`**, and `opencv-rust` SGBM becomes a **benchmark-only** quality baseline confined to the unpublished `vision-calibration-bench` crate — never a published crate, never the shipped impl. Validation uses **existing calibration-target data**: a C4-rectified stereo pair has known target-plane depth (ground truth from the calibration itself).

## Harness (`vision-calibration-bench::dense`, pure Rust, zero new deps)

- `DenseMatcher` trait (rectified pair → `DisparityMap`); `GrayBuffer`, `DisparityMap`, `DenseMatchOptions`.
- `synthetic_rectified_pair`: deterministic slanted-plane fixture — the right image is a bilinear warp of the left by the GT disparity, so an oracle recovers GT exactly and a real matcher can be scored unambiguously.
- `evaluate`: RMS / MAE / bad-pixel-rate / density over jointly-valid pixels (handles the zero-overlap edge case).
- `OracleMatcher` for plumbing tests. **8 tests.**

## Why no OpenCV in this PR

The project's quality gate is `cargo test --workspace --all-features`. OpenCV isn't installable in CI or the authoring environment, so adding an `opencv` cargo feature would break `--all-features` everywhere. The SGBM baseline therefore plugs into the `DenseMatcher` trait in a follow-up — from a workspace-excluded crate or a dedicated OpenCV CI job. The harness is ready for it (and for the pure-Rust matcher).

## Verification

`cargo build/clippy(-D warnings)/test/doc --workspace --all-features` green; bench `Cargo.toml` unchanged (zero new deps).

## Follow-ups (backlog C5)

OpenCV SGBM baseline (OpenCV-equipped env), then the pure-Rust matcher (block-matching MVP → SGM), benchmarked against the baseline + target-plane ground truth.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/vision-calibration-bench/src/dense.rs [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+
+    // --- right image: bilinear warp of left by GT disparity ---
+    let mut right_data = vec![0u8; width * height];
+    for y in 0..height {
+        for x in 0..width {
+            let d = gt.get(x, y);
+            let val = if d.is_finite() {
+                // right pixel at (x,y) corresponds to left pixel at (x - d, y).
+                let src_xf = x as f64 - d as f64;
+                left.bilinear(src_xf, y as f64).round() as u8
```
[the remark this answers is not in the document]
Comment (VitalyVorobyev) [reply crates/vision-calibration-bench/src/dense.rs]: Fixed in the latest commit. You are right — the fixture built `right[x] = left[x - d]` (disparity -d) while `gt` stored +d, so a convention-correct matcher would score as wrong. Now inverse-warps `right[x] = left[x + d]` so `right[x - d] == left[x]` (convention `d = x_left - x_right`), and synthesizes every right pixel from the raw plane disparity rather than gating on gt's left-frame occlusion NaNs. Replaced the self-consistency test (which had baked in the wrong sign) with `synthetic_pair_follows_disparity_sign_convention`, asserting `left[x] == right[x - d]` exactly under constant disparity — that test fails on the old warp.
